### PR TITLE
Adding Java documentation

### PR DESCRIPTION
**Pull request by andrewoh531**

### CODE REVIEW

File: notify/docs/sending-texts.md [hunk 1]
```diff
@@ -49,6 +50,14 @@ For example:
     "amount_owing": "$15.50"
 }
 ```
+---
+
+`reference`
+
+**Optional**
+
+A unique identifier you create. This reference identifies a single unique notification or a batch of notifications. If you do not have a reference, you must pass in an empty string or null.
```
Comment (jonathaningram) [inline notify/docs/sending-texts.md]: "empty string" and "null" are language dependent.

I wonder if we just word this like "if you do not have a reference, you can omit this argument".

File: notify/docs/listing-templates.md [hunk 2]
```diff
@@ -9,9 +9,23 @@ be sent and what personalisation variables each template requires.
...
 
+
+### Arguments
+---
+`template_type`
+
+**Optional**
+
+If omitted, the method returns all templates. Otherwise you can filter by:
+- email
```
Comment (jonathaningram) [inline notify/docs/listing-templates.md]: Should these be in code blocks to indicate they are values? I originally read this as a sentence: "you can filter by email/sms"

File: notify/docs/listing-templates/list.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+TemplateList allTemplates = client.getAllTemplates(null);
```
Comment (jonathaningram) [inline notify/docs/listing-templates/list.java:2]: You wrote above "If omitted, the method returns all templates", but then you're passing `null` in the example?

File: notify/docs/sending-texts.md (hunk 1 above)
Comment (andrewoh531) [reply notify/docs/sending-texts.md]: How about I leave the last sentence out altogether? "Omit" to me means leaving it out completely which is not valid for the java client.

File: notify/docs/listing-templates.md (hunk 2 above)
Comment (andrewoh531) [reply notify/docs/listing-templates.md]: I've reworded this

File: notify/docs/listing-templates/list.java (hunk 3 above)
Comment (andrewoh531) [reply notify/docs/listing-templates/list.java:2]: I've reworded general doc that accompanies this

File: notify/docs/sending-texts.md (hunk 1 above)
Comment (jonathaningram) [reply notify/docs/sending-texts.md]: Sounds good. The thing is marked as `Optional` anyway.

**Review by jonathaningram — APPROVED** [review]

**Pull request merged**